### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by over-geek**
## Recipe App: Final Changes

In this milestone, these were the features implemented:
- A login page.
- A registration page.
- A food list (with all CRUD implementation, except for 'update').
- A recipes list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.

collaborators: @francksefu @over-geek

### CODE REVIEW

File: app/views/recipes/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        <p> <%= @recipe.description %> </p>
+      </div>
+    </div>
+    <div>
+      <% if @recipe.user_id == current_user.id %>
+        <%= form_with model: @recipe, url: toggle_recipe_path(@recipe), method: :patch, local: true do |form| %>
+          <% new_public_status = !@recipe.is_public %>
+          <button type="submit" class="toggle-button <%= @recipe.is_public ? 'private' : 'public' %>">
+            <%= new_public_status ? 'status: Private' : 'status: Public' %>
+          </button>
```
Comment (Strangeal) [inline app/views/recipes/show.html.erb:18]: - [ ] Great job on your capstone project's UI. it's truly impressive! 😊 However, there's a small adjustment I'd recommend to ensure that your design aligns with the wireframe provided in the project requirements.

   - When examining your recipe details page, you'll notice that the button used to switch a recipe between `public` and `private` doesn't match the wireframe's design. In the wireframe, the button is a toggle, more like a switch, but in your current design, it's not. I'd suggest considering this change, as using a toggle button would make it more intuitive for users to understand that they can switch between making a recipe `public` and `private`. 🔄🔒 See below for more details:
   
<table>
<tr>
<th align=“middle”>Wireframe</th>
<th align=“middle”>Your design</th>
</tr>
<tr>
<td>

</td>
<td>
Screenshot 2023-10-20 at 22 15 10
</td>
</tr>
</table>

This adjustment should not only enhance the visual consistency with the project requirements but also improve the overall user experience. 

Happy coding! 😊👨‍💻🚀

File: app/views/recipes/public_recipes.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+<% @public_recipes.each do |recipe| %>
+  <div class="public_recipe_list">
+    <h3><%= link_to recipe.name, recipe_path(recipe, recipe) %></h3>
+    <p>Author: <%= recipe.user.name %></p>
+    <p>Total Food Items: <%= recipe.recipe_foods.count %></p>
+   <p>Total Amount: $<%= recipe.calculate_total_amount %></p>
+  </div>
+  <hr>
+<% end %>
```
Comment (Strangeal) [inline app/views/recipes/public_recipes.html.erb:11]: - [ ] Great job on creating such an engaging and visually appealing UI for your public recipe page! 👏 However, I noticed that your current design deviates from the wireframe provided in the project requirements. To ensure your design aligns with the wireframe and project specifications, I recommend making some adjustments. See below for more details on design differences:

<table>
<tr>
<th align=“middle”>Original design</th>
<th align=“middle”>Your design</th>
</tr>
<tr>
<td>

</td>
<td>
Screenshot 2023-10-20 at 22 22 17
</td>
</tr>
</table>

Aligning your design with the wireframe not only ensures consistency but also demonstrates your keen attention to detail. 🧐
Happy coding! 😊👨‍💻🚀

File: app/controllers/foods_controller.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  end
+
+  def set_food
+    @food = current_user.foods.find(params[:id])
+  end
+
+  def food_params
+    params.require(:food).permit(:name, :measurement_unit, :price)
+  end
+end
```
Comment (Strangeal) [inline app/controllers/foods_controller.rb:42]: - [ ] Excellent work in defining your controller action methods! 👍 However, I recommend marking the highlighted methods as `private` level methods. This practice enhances the encapsulation of your code and ensures that these methods are not accessible from outside the class. 

Happy coding! 😊👏

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @over-geek and @francksefu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 
- Followed *best-practices* ✅ 
- Well structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shella12 — APPROVED** [review]
Hi @over-geek and @francksefu, 


## Status : APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Made required methods private ✔️ 
- Added toggle button for public recipes ✔️ 
- Implemented all requirements 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**